pkg/model/manifest_friendly_err.go

```diff
@@ -0,0 +1,165 @@
+		suggest.NewStrReplaceRule("in type model.ManifestBuild", "the 'build' section"),
+		suggest.NewStrReplaceRule("into model.ManifestBuild", "into a 'build' object"),
+		suggest.NewStrReplaceRule("in type model.buildInfoRaw", "the 'build' object"),
+		suggest.NewStrReplaceRule("in type model.devType", "the 'dev' object"),
+		suggest.NewStrReplaceRule("into model.devType", "the 'dev' object"),
+
+		// yaml types
+		suggest.NewStrReplaceRule(yaml.NodeTagSeq, "list"),
+		suggest.NewStrReplaceRule(yaml.NodeTagString, "string"),
+		suggest.NewStrReplaceRule(yaml.NodeTagBool, "boolean"),
+		suggest.NewStrReplaceRule(yaml.NodeTagInt, "integer"),
+		suggest.NewStrReplaceRule(yaml.NodeTagFloat, "float"),
+		suggest.NewStrReplaceRule(yaml.NodeTagMap, "object"),
+
+		// misc
+		suggest.NewStrReplaceRule("yaml: unmarshal errors:\n", ""),
+		suggest.NewStrReplaceRule("invalid manifest: ", ""),
+		indentNumLines(),
+	)
+
+	return rules
+}
+
+func isYamlErrorWithoutLinkToDocs(err error) bool {
+	return isYamlError(err) && !hasLinkToDocs(err)
+}
+
+// hasLinkToDocs checks if the error already contains a link to our docs
+func hasLinkToDocs(err error) bool {
+	return strings.Contains(err.Error(), "https://www.okteto.com/docs")
+}
+
+// isYamlError check if the error is related to YAML unmarshalling
+func isYamlError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// detect yaml errors by prefix
+	if strings.Contains(err.Error(), "yaml:") {
+		return true
+	}
+	// detect yaml errors by type
+	var typeError *yaml.TypeError
+	return errors.As(err, &typeError)
+}
+
+func addYamlParseErrorHeading() *suggest.Rule {
+	addUrl := func(e error) error {
+		errorWithUrlToDocs := fmt.Sprintf("Your okteto manifest is not valid, please check the following errors:\n%s", e.Error())
+		return errors.New(errorWithUrlToDocs)
+	}
+
+	return suggest.NewRule(isYamlErrorWithoutLinkToDocs, addUrl)
+}
+
+// addUrlToManifestDocs appends to the error the URL to the Okteto manifest ref docs
+func addUrlToManifestDocs() *suggest.Rule {
+	addUrl := func(e error) error {
+		docsURL := "https://www.okteto.com/docs/reference/manifest"
+		errorWithUrlToDocs := fmt.Sprintf("%s\n    Check out the okteto manifest docs at: %s", e.Error(), docsURL)
+		return errors.New(errorWithUrlToDocs)
+	}
+
+	return suggest.NewRule(isYamlErrorWithoutLinkToDocs, addUrl)
+}
+
+func indentNumLines() *suggest.Rule {
+	pattern := `(?:yaml: )?line (\d+):`
+	re := regexp.MustCompile(pattern)
+
+	condition := func(e error) bool {
+		return re.MatchString(e.Error())
+	}
+
+	transformation := func(e error) error {
+		newErr := re.ReplaceAllString(e.Error(), "   - line $1:")
+		return errors.New(newErr)
+	}
+
+	return suggest.NewRule(condition, transformation)
+}
+
+// fieldsNotExistingRule replaces "not found" fields which are unknown to the Okteto manifest specification
+func fieldsNotExistingRule() *suggest.Rule {
+	pattern := `field (\w+) not found`
+	re := regexp.MustCompile(pattern)
```

Comment:
https://github.com/okteto/okteto/pull/4009#discussion_r1361118090 this was already addressed before I think

Reply:
as per the comment below about the fixed pattern